socs/agents/srs_cg635m/drivers.py

```diff
@@ -0,0 +1,129 @@
+             2 = 2.5V
+             3 = 3.3V (The default for our current setup)
+             4 = 5.0V
+        """
+
+        self.write("STDC?")
+        stdc = self.read()
+
+        return int(stdc)
+
+    def get_runs(self):
+        """
+        Queries the clock for the current Running State (RUNS).
+
+        Returns an int which represents the following running states:
+            0 = Not Running (Output is off)
+            1 = Running (Output is on)
+        """
+
+        self.write("RUNS?")
+        runs = self.read()
+
+        return int(runs)
+
+    def get_lock_statuses(self):
+        """
+        Queries the clock for the current Lock Registers (LCKR).
+
+        The lock registers represent the current Lock status for following registers:
+            RF_UNLOCK
+            19MHZ_UNLOCK
+            10MHZ_UNLOCK
+            RB_UNLOCK
+            OUT_DISABLED
+            PHASE_SHIFT
+
+        Returns a dict of the registers and registers statuses with the keys being the registers
+            and the values being an int representing the register statuses.
+            1 = True, 0 = False
+        """
+        self.write("LCKR?")
+        lckr = self.read()
+
+        # The LCKR is a 8 bit register with each register status represented by a single bit.
+        # The LCKR? query returns a single int representation of the register bits
+        # The decode_lckr function finds the register bit for all registers
+        lckr_status = self._decode_lckr(lckr)
+
+        return lckr_status
+
+    def _decode_lckr(self, lckr):
+        """
+        Takes the int representation of the lock register (lckr) and translates it into dict form.
+        The dict keys are the register names and the values are the register status:
+            1 = True, 0 = False
+
+        The incoming lckr int should always be <256 because its a int representation of an 8 bit reigster.
+
+        The lock register bits are as follows:
+            0 = RF_UNLOCK
+            1 = 19MHZ_UNLOCK
+            2 = 10MHZ_UNLOCK
+            3 = RB_UNLOCK
+            4 = OUT_DISABLED
+            5 = PHASE_SHIFT
+            6 = Reserved
+            7 = Reserved
+        """
+
+        registers = {"RF_UNLOCK": None,
+                     "19MHZ_UNLOCK": None,
+                     "10MHZ_UNLOCK": None,
+                     "RB_UNLOCK": None,
+                     "OUT_DISABLED": None,
+                     "PHASE_SHIFT": None}
+
+        try:
+            lckr = int(lckr)
+
+            if not 0 <= lckr <= 255:
+                # If the lckr register is outside of an 8 bit range
+                raise ValueError
+
+            # Decode the lckr int by performing successive int division and subtractionof 2**(5-i)
+            for i, register in enumerate(list(registers)[::-1]):
+                register_bit = int(lckr / (2**(5 - i)))
+                registers[register] = int(register_bit)
+                lckr -= register_bit * (2**(5 - i))
```

Comment:
I think you can do this as is but also could maybe do it a bit more compactly: `register_list = [bool(msg & 1 << (8-i)) for i in range(1,9)]` I tested this as follows:

```
In [64]: msg=int('01110000',2);print(msg)
112

In [65]: [bool(msg & 1 << (8-i)) for i in range(1,9)]
Out[65]: [False, True, True, True, False, False, False, False]

In [66]: msg=int('10000000',2);print(msg)
128

In [67]: [bool(msg & 1 << (8-i)) for i in range(1,9)]
Out[67]: [True, False, False, False, False, False, False, False]

In [68]: msg=int('00000001',2);print(msg)
1

In [69]: [bool(msg & 1 << (8-i)) for i in range(1,9)]
Out[69]: [False, False, False, False, False, False, False, True]
```

Then you can define `registers = {rn: rl for r, rl in zip(reg_names, register_list)}` where `reg_names` is a list of the register key names that you've defined in the dict in line 106.

Reply:
I knew there was probably a much cleaner way to do this that one of the friendly daq reviewers would surely suggest to me. :)

 I can switch this decoding to this way for clarity sake. Though I'd like to keep them as ints so they're easily displayable on grafana, much like the leak detectors.